File: src/Application Framework/Forms And Controls/Global/FRMGenDropBox.vb
Public Class FRMGenDropBox

    Public IncDD As Boolean = True

    Private Sub btnOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOK.Click
        If cbo1.Visible And Combo.GetSelectedItemValue(cbo1) = 0 Then
            ShowMessage("Please select a Valid Option", MessageBoxButtons.OK, MessageBoxIcon.Error)
        ElseIf cbo2.Visible And Combo.GetSelectedItemValue(cbo2) = 0 Then
            ShowMessage("Please select a Valid Option", MessageBoxButtons.OK, MessageBoxIcon.Error)
        ElseIf txtref.Visible And txtref.Text.Length = 0 Then
            ShowMessage("Please enter a payment reference", MessageBoxButtons.OK, MessageBoxIcon.Error)
        Else
            DialogResult = DialogResult.OK
            If Me.Modal Then
                Me.Close()
            Else
                Me.Dispose()
            End If
        End If
    End Sub

    Private Sub FRMChangePriority_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

        Combo.SetUpCombo(Me.cbo1, DynamicDataTables.ContractInitialPaymentTypes, "ValueMember", "DisplayMember", Entity.Sys.Enums.ComboValues.Please_Select)
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Combo.SetSelectedComboItem_By_Value(cbo1, "0")
        If Me.Modal Then
            Me.Close()
        Else
            Me.Dispose()
        End If
    End Sub

End Class

File: src/Application Framework/Forms And Controls/Global/FRMGenDropBox.cs
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Windows.Forms;

namespace FSM
{
    public partial class FRMGenDropBox
    {
        public FRMGenDropBox()
        {
            InitializeComponent();
        }

        public bool IncDD = true;

        private void btnOK_Click(object sender, EventArgs e)
        {
            if (cbo1.Visible & Conversions.ToDouble(Combo.get_GetSelectedItemValue(cbo1)) == 0)
            {
                App.ShowMessage("Please select a Valid Option", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            else if (cbo2.Visible & Conversions.ToDouble(Combo.get_GetSelectedItemValue(cbo2)) == 0)
            {
                App.ShowMessage("Please select a Valid Option", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            else if (txtref.Visible & txtref.Text.Length == 0)
            {
                App.ShowMessage("Please enter a payment reference", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            else
            {
                DialogResult = DialogResult.OK;
                if (Modal)
                {
                    Close();
                }
                else
                {
                    Dispose();
                }
            }
        }

        private void FRMChangePriority_Load(object sender, EventArgs e)
        {
            var argc = cbo1;
            Combo.SetUpCombo(ref argc, DynamicDataTables.ContractInitialPaymentTypes, "ValueMember", "DisplayMember", Entity.Sys.Enums.ComboValues.Please_Select);
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            var argcombo = cbo1;
            Combo.SetSelectedComboItem_By_Value(ref argcombo, "0");
            if (Modal)
            {
                Close();
            }
            else
            {
                Dispose();
            }
        }
    }
}